app/controllers/template_invocations_controller.rb

```diff
@@ -23,6 +23,30 @@ def show
     @line_counter = params[:line_counter].to_i
   end
 
+  def list_template_invocations_task_by_hosts
+    hosts = Host.search_for(params[:host_query])
+    template_invocations_task_by_hosts = {}
+    hosts.each do |host|
+      template_invocation = @job_invocation.template_invocations.find { |template_inv| template_inv.host_id == host.id }
+      next unless template_invocation
+      template_invocation_task = template_invocation.run_host_job_task
+      template_invocations_task_by_hosts[host.id] =
+        {
+          :task_id => template_invocation_task.id,
+          :task_cancellable => template_invocation_task.cancellable?,
+          :permissions => {
+            :view_foreman_tasks => User.current.allowed_to?(:view_foreman_tasks),
```

Comment:
We could be more specific and instead of asking "Can the user view tasks in general?" we could ask "Can the user view this host's task?". 

```suggestion
            :view_foreman_tasks => User.current.can?(:view_foreman_tasks, template_invocation_task,
```

Alternatively, in the old ui we use this
```
authorized_for(:permission => :view_foreman_tasks, :auth_object => task)
```